### ISSUE TITLE
NadamOptimizer throws InvalidArgumentError (incompatible shapes)

### ISSUE THREAD

**Issue by strubell** [tensorflow/tensorflow#13980]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes, but this issue can be replicated by changing one line in the example script `word2vec_basic.py`
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux CentOS 7, MacOS High Sierra 
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: ('v1.4.0-rc0-21-g1e25994', '1.4.0-rc1'). I also know someone who encountered this in the 1.3 release.
- **Python version**: 2.7.10, 2.7.14
- **Bazel version (if compiling from source)**: n/a
- **CUDA/cuDNN version**: 8.0 / 6.0
- **GPU model and memory**: Titan X 12gb
- **Exact command to reproduce**:
Change line 190 in `tensorflow/examples/tutorials/word2vec/word2vec_basic.py` to:
`optimizer = tf.contrib.opt.NadamOptimizer(1.0).minimize(loss)`. 
Run.

### Describe the problem
When I attempt to train a model with `tf.controb.opt.NadamOptimizer`, TensorFlow crashes with an InvalidArgumentError (Incompatible shapes: [105] vs. [50000]), which is thrown from the optimizer (in my code it's when trying to apply updates to the word embedding table.) It looks like it's trying to apply sparse updates to the embeddings using dense operations, causing a shape mismatch, or something along those lines. `AdamOptimizer` and `LazyAdamOptimizer` work fine.

### Source code / logs
Traceback can be found [here](https://gist.github.com/strubell/37897084888252989750b58260f76ff5).
The error is easily replicable in the example script `word2vec_basic.py`.

**Comment by skye** [comment]
@alisidd can you look into this?

**Comment by alisidd** [comment]
@skye sorry for the delayed response. Yes, I can look into it.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 140 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 156 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 171 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 186 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 201 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 216 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 231 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 246 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 261 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 276 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 291 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rohan100jain** [comment]
Seems like there is a PR that should fix this issue. Lets wait for that to go in.

**Comment by wt-huang** [comment]
@strubell For `tf.contrib.opt.NadamOptimizer` you would need to modify more code base to make it function.  In this case you can use `optimizer = tf.train.AdamOptimizer(0.01).minimize(loss)`, which should work and serve the similar purpose.
